File: leetcode/14_trees/94_Binary_Tree_Inorder_Traversal.py
class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right


class Solution:
    def inorderTraversal(self, root: Optional[TreeNode]) -> List[int]:
        def _helper(root, data):
            if not root:
                return
            _helper(root.left, data)
            data.append(root.val)
            _helper(root.right, data)

        data = []
        _helper(root, data)
        return data

File: leetcode/14_trees/94_Binary_Tree_Inorder_Traversal.go

type TreeNode struct {
	Val   int
	Left  *TreeNode
	Right *TreeNode
}

func inorderTraversal(root *TreeNode) []int {
	var data []int

	var helper func(*TreeNode, *[]int)

	helper = func(root *TreeNode, data *[]int) {
		if root == nil {
			return
		}
		helper(root.Left, data)
		*data = append(*data, root.Val)
		helper(root.Right, data)
	}

	helper(root, &data)
	return data
}
